File: Sismique/DTO/ListReport.cs
using ARIO.Models;
using MetadataExtractor;
using System;

namespace ARIO.DTO
{
    public class ListReport
    {
        public int ID { get; set; }

        public string Name { get; set; }

        public DateTime Date { get; set; }

        public int NumberImages { get; set; }

        public string Thumbnail { get; set; }

        public double Latitude { get; set; }

        public double Longitude { get; set; }
        public bool Shared { get; set; }
        public static ListReport FromReport(Report report, int numberImages, Image thumbnailImage, GeoLocation location)
        {
            return new ListReport
            {
                ID = report.ID,
                Name = report.Name,
                Date = report.Date,
                Shared = report.Shared,
                NumberImages = numberImages,
                Thumbnail = thumbnailImage?.Thumbnail,
                Latitude = location.Latitude,
                Longitude = location.Longitude
            };
        }

        public bool HasGpsCoordinates()
        {
            return Latitude != 0 && Longitude != 0;
        }
    }
}

File: Sismique/ClientApp/app/components/shared/listreport.ts
export interface IListReportProperties {
    id: number;
    name: string;
    date: Date;
    shared: boolean;
    numberImages: number;
    thumbnail: string;
    latitude: number;
    longitude: number;
}

export class ListReport {
    id: number;
    name: string;
    date: Date;
    shared: boolean;
    numberImages: number;
    thumbnail: string;
    latitude: number;
    longitude: number;

    deserialize(input: IListReportProperties): ListReport {
        this.id = input.id;
        this.name = input.name;
        this.date = new Date(input.date);
        this.shared = input.shared;
        this.numberImages = input.numberImages;
        this.thumbnail = input.thumbnail;
        this.latitude = input.latitude;
        this.longitude = input.longitude;

        return this;
    }

    hasGpsCoordinates(): boolean {
        return this.latitude !== 0.0 && this.longitude !== 0.0;
    }
}
